app/queries/servers/custom_profile.ts

```diff
@@ -0,0 +1,177 @@
+
+/**
+ * Get a custom profile field by ID
+ * @param database - The database instance
+ * @param fieldId - The field ID
+ * @returns The custom profile field or undefined
+ */
+export const getCustomProfileFieldById = async (database: Database, fieldId: string) => {
+    try {
+        const fieldRecord = await database.get<CustomProfileFieldModel>(CUSTOM_PROFILE_FIELD).find(fieldId);
+        return fieldRecord;
+    } catch {
+        return undefined;
+    }
+};
+
+/**
+ * Observe a custom profile field
+ * @param database - The database instance
+ * @returns Observable of the custom profile field
+ */
+export const observeCustomProfileFields = (database: Database) => {
+    return queryCustomProfileFields(database).observeWithColumns(['update_at', 'delete_at']).pipe(
+        distinctUntilChanged(),
+    );
+};
+
+/**
+ * Get a custom profile attribute by field ID and user ID
+ * @param database - The database instance
+ * @param fieldId - The field ID
+ * @param userId - The user ID
+ * @returns The custom profile attribute or undefined
+ */
+export const getCustomProfileAttribute = async (database: Database, fieldId: string, userId: string) => {
+    try {
+        const attributeRecord = await database.get<CustomProfileAttributeModel>(CUSTOM_PROFILE_ATTRIBUTE).query(
+            Q.where('field_id', fieldId),
+            Q.where('user_id', userId),
+            Q.take(1),
+        ).fetch();
+        return attributeRecord[0];
+    } catch {
+        return undefined;
+    }
+};
+
+/**
+ * Observe a custom profile attribute
+ * @param database - The database instance
+ * @param fieldId - The field ID
+ * @param userId - The user ID
+ * @returns Observable of the custom profile attribute
+ */
+export const observeCustomProfileAttribute = (database: Database, fieldId: string, userId: string) => {
+    return database.get<CustomProfileAttributeModel>(CUSTOM_PROFILE_ATTRIBUTE).query(
+        Q.where('field_id', fieldId),
+        Q.where('user_id', userId),
+        Q.take(1),
+    ).observe().pipe(
+        switchMap((result) => (result.length ? result[0].observe() : of$(undefined))),
+    );
+};
+
+/**
+ * Observe all custom profile attributes for a user
+ * @param database - The database instance
+ * @param userId - The user ID
+ * @returns Observable of the custom profile attributes
+ */
+export const observeCustomProfileAttributesByUserId = (database: Database, userId: string): Observable<CustomProfileAttributeModel[]> => {
+    return queryCustomProfileAttributesByUserId(database, userId).observeWithColumns(['value']).pipe(distinctUntilChanged());
+};
+
+/**
+ * Convert custom profile attributes to the UI-ready CustomAttribute format
+ * @param database - The database instance
+ * @param attributes - Array of custom profile attribute models
+ * @returns Promise resolving to array of formatted CustomAttribute objects
+ */
+export const convertProfileAttributesToCustomAttributes = async (
```

Comment:
I would love to see a benchmark of some sort for this function, seems that it can be rather slow

Reply:
I'll push the test so people can run it, but so looks like I'm getting 1ms for 200 records and that's 10 times the max allowed.
```
📢 Log: Testing conversion of 200 attributes

📢 Log: Time with sorting: 1.00ms
```